src/llscan.cc

```diff
@@ -0,0 +1,497 @@
+    : target_(target), mapstoinstances_(mapstoinstances) {
+  found_count_ = 0;
+  address_byte_size_ = target_.GetProcess().GetAddressByteSize();
+  // Load V8 constants from postmortem data
+  llv8.Load(target);
+}
+
+
+/* Visit every address, a bit brute force but it works. */
+uint64_t FindJSObjectsVisitor::Visit(uint64_t location, uint64_t available) {
+  lldb::SBError error;
+
+  // Test if the map points to a real map.
+  // Try to create an object out of it.
+
+  uint64_t word = target_.GetProcess().ReadUnsignedFromMemory(
+      location, target_.GetProcess().GetAddressByteSize(), error);
+
+  v8::Value v8_value(&llv8, word);
+
+  v8::Error err;
+  // Test if this is SMI
+  // Skip inspecting things that look like Smi's, they aren't objects.
+  v8::Smi smi(v8_value);
+  if (smi.Check()) {
+    return address_byte_size_;
+  }
+
+  v8::HeapObject heap_object(v8_value);
+  if (!heap_object.Check()) {
+    return address_byte_size_;
+  }
+  if (heap_object.IsHoleOrUndefined(err)) {
+    return address_byte_size_;
+  }
+  if (err.Fail()) {
+    return address_byte_size_;
+  }
+
+  v8::HeapObject map_object = heap_object.GetMap(err);
+  if (err.Fail() || !map_object.Check()) {
+    return address_byte_size_;
+  }
+
+  if (!IsAHistogramType(heap_object, err)) {
+    return address_byte_size_;
+  }
+
+  if (err.Fail()) {
+    return address_byte_size_;
+  }
+
+  v8::Map map(map_object);
+
+  v8::Value::InspectOptions inspect_options;
+  inspect_options.detailed = false;
+  inspect_options.print_map = false;
+  inspect_options.string_length = 0;
+
+  /* No entry in the map, create a new one. */
+  if (mapstoinstances_.count(map_object.raw()) == 0) {
+    TypeRecord* t = new TypeRecord();
+    t->map = map_object.raw();
+    t->instance_count++;
+    // TODO(indutny) - Is this correct or is there a better way to find the size
+    // of one instance of an object?
+    t->total_instance_size = map.InstanceSize(err);
+
+    t->type_name =
+        new std::string(GetTypeName(heap_object, inspect_options, err));
+
+    mapstoinstances_.insert(
+        std::pair<uint64_t, TypeRecord*>(map_object.raw(), t));
+    t->instances = new std::set<uint64_t>();
+  } else {
+    /* Update an existing instance, if we haven't seen this one before. */
+    TypeRecord* t = mapstoinstances_.at(map_object.raw());
+    /* Determine if this is a new instance.
+     * (We are scanning pointers to objects, we may have seen this location
+     * before.)
+     */
+    if (t->instances->count(word) == 0) {
+      t->instances->insert(word);
+      t->instance_count++;
+      // TODO(indutny) - Is this correct or is there a better way to find the
+      // size of one instance of an object?
+      t->total_instance_size += map.InstanceSize(err);
+    }
+  }
+
+  if (err.Fail()) {
+    return address_byte_size_;
+  }
+
+  found_count_++;
+
+  /* Just advance one word.
+   * (Should advance by object size, assuming objects can't overlap!)
```

Comment:
This will probably not work for now. The core file segment size is not necessarily a start of the heap, and if we will skip more than a pointer, we may eventually skip a lot of objects.